symbolic-base/test/Tests/Symbolic/Data/Common.hs

```diff
@@ -0,0 +1,234 @@
+  , Context (x (Interpreter a)) ~ Interpreter a
+  , Layout (x (Interpreter a)) ~ ix
+  , SymbolicInput (y (Interpreter a))
+  , Context (y (Interpreter a)) ~ Interpreter a
+  , Layout (y (Interpreter a)) ~ iy
+  , SymbolicOutput (z (Interpreter a))
+  , Context (z (Interpreter a)) ~ Interpreter a
+  , Layout (z (Interpreter a)) ~ o
+  , Payload (z (Interpreter a)) ~ U1
+  , i ~ ix :*: iy
+  ) => ArithmeticCircuit a i o -> x (Interpreter a) -> y (Interpreter a) -> z (Interpreter a)
+evalCircuit2 ac x y =
+  restore $ const (Interpreter $ eval ac input, U1)
+  where
+    ix = runInterpreter $ arithmetize x Proxy
+    iy = runInterpreter $ arithmetize y Proxy
+    input = ix :*: iy
+
+compileCircuit0 :: forall a x o . (Binary a
+  , SymbolicInput (x (ArithmeticCircuit a U1))
+  , Context (x (ArithmeticCircuit a U1)) ~ ArithmeticCircuit a U1
+  , Layout (x (ArithmeticCircuit a U1)) ~ o
+  , Payload (x (ArithmeticCircuit a U1)) ~ U1
+  ) => (forall c . Symbolic c => x c) -> ArithmeticCircuit a U1 o
+compileCircuit0 v =
+  compileWith @a id (\U1 -> (U1, U1)) $ v @(ArithmeticCircuit a U1)
+
+compileCircuit1 :: forall a x y i o . (Binary a, Representable i
+  , SymbolicInput (x (ArithmeticCircuit a i))
+  , Context (x (ArithmeticCircuit a i)) ~ ArithmeticCircuit a i
+  , Layout (x (ArithmeticCircuit a i)) ~ i
+  , Payload (x (ArithmeticCircuit a i)) ~ U1
+  , SymbolicOutput (y (ArithmeticCircuit a i))
+  , Context (y (ArithmeticCircuit a i)) ~ ArithmeticCircuit a i
+  , Layout (y (ArithmeticCircuit a i)) ~ o
+  ) => (forall c . Symbolic c => x c -> y c) -> ArithmeticCircuit a i o
+compileCircuit1 func =
+  compileWith @a id (\i -> (U1 :*: U1, i :*: U1)) $ func @(ArithmeticCircuit a i)
+
+compileCircuit2 :: forall a x y z ix iy i o . (Binary a, Representable i
+  , SymbolicInput (x (ArithmeticCircuit a i))
+  , Context (x (ArithmeticCircuit a i)) ~ ArithmeticCircuit a i
+  , Layout (x (ArithmeticCircuit a i)) ~ ix
+  , Payload (x (ArithmeticCircuit a i)) ~ U1
+  , SymbolicInput (y (ArithmeticCircuit a i))
+  , Context (y (ArithmeticCircuit a i)) ~ ArithmeticCircuit a i
+  , Layout (y (ArithmeticCircuit a i)) ~ iy
+  , Payload (y (ArithmeticCircuit a i)) ~ U1
+  , SymbolicOutput (z (ArithmeticCircuit a i))
+  , Context (z (ArithmeticCircuit a i)) ~ ArithmeticCircuit a i
+  , Layout (z (ArithmeticCircuit a i)) ~ o
+  , i ~ ix :*: iy
+  ) => (forall c . Symbolic c => x c -> y c -> z c) -> ArithmeticCircuit a i o
+compileCircuit2 func =
+  compileWith @a id (\(ix :*: iy) -> (U1 :*: U1 :*: U1, ix :*: iy :*: U1)) $ func @(ArithmeticCircuit a i)
+
+type SymbolicFunction0 x = forall c . Symbolic c
+  => x c
+
+specSymbolicFunction0 :: forall a x o .
+  ( Show a
+  , Eq (x (Interpreter a))
+  , Show (x (Interpreter a))
+  , MatchingSymbolicInput x o (Interpreter a) (ArithmeticCircuit a U1)
+  ) => String -> SymbolicFunction0 x -> Spec
+specSymbolicFunction0 desc v = describe desc $ do
+  it "evaluates correctly" $ evalCircuit0 @a (compileCircuit0 v) === v
```

Comment:
This property is always satisfied because each of the contexts evaluates the same witnesses in the same way thanks to the way `Symbolic` interface is defined